src/cli/self_update/windows.rs

```diff
@@ -460,35 +463,39 @@ pub(crate) fn do_add_to_path(process: &Process) -> Result<()> {
     do_add_to_programs(process)
 }
 
-fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
+fn _apply_new_path(new_path: Option<String>) -> Result<()> {
     use std::ptr;
     use windows_sys::Win32::Foundation::*;
     use windows_sys::Win32::UI::WindowsAndMessaging::{
-        SendMessageTimeoutA, HWND_BROADCAST, SMTO_ABORTIFHUNG, WM_SETTINGCHANGE,
+        SendMessageTimeoutW, HWND_BROADCAST, SMTO_ABORTIFHUNG, WM_SETTINGCHANGE,
     };
 
     let new_path = match new_path {
         Some(new_path) => new_path,
         None => return Ok(()), // No need to set the path
     };
 
-    let root = RegKey::predef(HKEY_CURRENT_USER);
-    let environment = root.open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)?;
+    // let root = RegKey::predef(HKEY_CURRENT_USER);
+    // let environment = root.open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)?;
+
+    let environment = CURRENT_USER.create("Environment")?;
 
     if new_path.is_empty() {
-        environment.delete_value("PATH")?;
+        environment.remove_value("PATH")?;
     } else {
-        let reg_value = RegValue {
-            bytes: to_winreg_bytes(new_path),
-            vtype: RegType::REG_EXPAND_SZ,
-        };
-        environment.set_raw_value("PATH", &reg_value)?;
+        // let reg_value = RegValue {
+        //     bytes: to_winreg_bytes(new_path),
+        //     vtype: RegType::REG_EXPAND_SZ,
+        // };
+        // environment.set_raw_value("PATH", &reg_value)?;
+
+        environment.set_string("PATH", &new_path)?;
     }
 
     // Tell other processes to update their environment
     #[allow(clippy::unnecessary_cast)]
     unsafe {
-        SendMessageTimeoutA(
+        SendMessageTimeoutW(
```

Comment:
Why are we switching this?

Reply:
I noticed that there is a function called `CreateFileW()` in `src/cli/self_update/windows.rs`, so I want to unify them. But according to what @rami3l said, I'll put it into a seperate PR.

Reply:
@djc *A is for ANSI string-based APIs ans *W for UTF16-based ones. In this particular function there seems to be no differences between the two but I guess in terms of consistency *W is still slightly better: https://stackoverflow.com/a/7424550

Reply:
@rami3l Since this PR was merged, should I still make this change and create a PR?